Send Get Request To Server, Server Responds 'Hello'

DeKaylaPhillips opened the pull request:
<!--
     Before submitting a Pull Request, please ensure you've done the following:
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.
-->
# Send Get Request To Server, Server Responds 'Hello'

## Type of PR <!-- (check all applicable) -->

- [X] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This branch refactors the app's `listen()` method and tells the server to listen on Port 3000 instead of the initial Port 5000. The branch is also more modular in file structure, adds a test suite with a mock test to test the behavior of the server and its response, and includes changes to configuration files to resolve dependency and parsing errors. 

## Related Tickets & Documents

<img width="573" alt="Screen Shot 2023-06-12 at 11 53 03 AM" src="https://github.com/DeKaylaPhillips/Echo-Server/assets/91583190/75dee42e-2ba3-4f32-bd91-52eebcf6cd79">

- Closes <https://trello.com/c/NNpgqAq8/32-when-i-send-a-get-request-to-the-server-the-server-responds-with-hello>

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

Commit:
Refactor server to listen on Port 3000

Commit:
Create workflow.yml

Commit:
Add/update config files & test suite

Commit:
Add index file

Refactored server, added test for server response, and removed server file test suite.

Commit:
Update project configs

Commit:
Refactor and fix linter errors

Commit:
Update project configs & clean up

Updated project configuration files, cleaned gitignore file and lint errors, and fixed imports.

Naomi-Dennis commented:
Remove `package-lock.json` file

Naomi-Dennis commented on .gitignore:
```diff
@@ -1,3 +1,5 @@
 /node_modules
-package-lock.json
\ No newline at end of file
+
```
Remove whitespace

DeKaylaPhillips replied:
Removed the whitespace here.

Naomi-Dennis commented on package.json:
```diff
@@ -4,19 +4,28 @@
+    "test": "jest",
+    "dev": "ts-node src/server.ts",
+    "lint": "eslint src/**/*.ts",
+    "lint tests": "eslint tests/**/*.spec.ts"
```
Good, but tests don't need linter checks!

DeKaylaPhillips replied:
Removed the lint script for the tests. :)

Naomi-Dennis commented on tests/controller/index.spec.ts:
```diff
@@ -0,0 +1,15 @@
+
+    indexController(mockRequest as Request, mockResponse as Response)
+
+    expect(mockResponse.send).toHaveBeenCalled()
+    expect(mockResponse.send).toHaveBeenCalledTimes(1)
```
Remove line 11

DeKaylaPhillips replied:
Line 11 has been removed. :)

Naomi-Dennis reviewed (approved):
🚢

Naomi-Dennis commented:
Clean up the commits. Try to get the commit count to either 8 or below. 
One way to do it is to use `git reset --soft HEAD~[# of commits]`

Commit:
Update project configs & refactor test

Removed unnecessary assertion in test file and lint script in project configurations. Updated gitignore and added package-lock for CI recognition.

DeKaylaPhillips commented:
> Remove `package-lock.json` file

The package-lock is required by the CI and was added to GH to run properly. :)
